service/api_task.go

```diff
@@ -547,9 +554,12 @@ func (as *APIServer) NextTask(w http.ResponseWriter, r *http.Request) {
 		gimlet.WriteJSON(w, response)
 		return
 	}
-
 	var agentExit bool
-	response, agentExit = handleOldAgentRevision(response, h, w, r)
+	details, agentExit := getDetails(response, h, w, r)
```

Comment:
I don't totally understand this refactor. Does this do anything except split `handleOldAgentRevision` into 2 functions?

Reply:
No it basically just splits handleOldAgentRevision into 2 functions, because before we only unmarshalled GetNextTaskDetails if we had to set a new agent, but now we need it to determine if the task group is finished.